### ISSUE TITLE
appComponent and React.createFactory

### ISSUE THREAD

**Issue by mbfisher** [yahoo/fluxible#73]
I've recently updated from 0.1.5 and received a strange error message from React, specifically `ReactClass#createClass`:

```
Uncaught TypeError: Cannot read property '__reactAutoBindMap' of null
```

I fixed this by wrapping my `appComponent` in `React.createFactory`:

```js
var App = require('App.jsx');
var app = new Fluxible({
    component: React.createFactory(App)
});
```

I have no idea why it didn't work nor why it now does! Do you understand what's happening? Have you had any other reports of this?

M

**Comment by mridgway** [comment]
React changed the way component classes are created in [React 0.12](http://facebook.github.io/react/blog/2014/10/28/react-v0.12.html#breaking-changes). This caused the API in Fluxible to become a little weird, but you're doing it correctly. I would like to add a warning here if someone passes a non-factory just to make it more user friendly.

**Comment by mbfisher** [comment]
I've had a go at a PR for this but can't find a way to differentiate between the 2 forms; any suggestions?

M

**Comment by mridgway** [comment]
Yeah, I was having issues too. You can theoretically look at `component.type` which would indicate that it is a factory, but React warns when you access this property.

**Comment by mridgway** [comment]
It looks like instead of needing a factory, we could just use `React.createElement(Component, props)` directly inside of our `Fluxible.createElement`. Let me make sure that this is backwards compatible and then we can make the change.

**Comment by mridgway** [comment]
Fixed in v0.3.1. Thanks for bringing this to our attention @mbfisher!
